src/plugins/data/common/index_patterns/fields/field_list.ts

```diff
@@ -58,8 +58,14 @@ export class FieldList extends Array<IndexPatternField> implements IIndexPattern
     this.groups.get(field.type)!.set(field.name, field);
   };
   private removeByGroup = (field: IFieldType) => this.groups.get(field.type)!.delete(field.name);
-  private calcDisplayName = (name: string) =>
-    this.shortDotsEnable ? shortenDottedString(name) : name;
+  private calcDisplayName = (name: string) => {
```

Comment:
Instead of looking up the name, pass it to the IndexPatternField instance. It should also be added to the FieldSpec.

Reply:
np, changed that